src/core/public/chrome/ui/header/recent_items.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import React, { useMemo, useState } from 'react';
+import * as Rx from 'rxjs';
+import {
+  EuiPopover,
+  EuiHeaderSectionItemButton,
+  EuiTextColor,
+  EuiListGroup,
+  EuiListGroupItem,
+  EuiTitle,
+  EuiIcon,
+} from '@elastic/eui';
+import useObservable from 'react-use/lib/useObservable';
+import { ChromeRecentlyAccessedHistoryItem } from '../..';
+import { WorkspaceObject } from '../../../workspace';
+
+export interface Props {
+  recentlyAccessed$: Rx.Observable<ChromeRecentlyAccessedHistoryItem[]>;
+  workspaceList$: Rx.Observable<WorkspaceObject[]>;
+  navigateToUrl: (url: string) => Promise<void>;
+}
+
+export const RecentItems = ({ recentlyAccessed$, workspaceList$, navigateToUrl }: Props) => {
+  const [isPopoverOpen, SetIsPopoverOpen] = useState(false);
+
+  const recentlyAccessedItems = useObservable(recentlyAccessed$, []);
+  const workspaceList = useObservable(workspaceList$, []);
+
+  const items = useMemo(() => {
+    // Only display five most latest items
+    return recentlyAccessedItems.slice(0, 5).map(({ link, label, workspaceId }) => {
+      return {
+        link,
+        label,
+        workspaceId,
+        workspaceName: workspaceList.find((workspace) => workspace.id === workspaceId)?.name ?? '',
+      };
+    });
+  }, [recentlyAccessedItems, workspaceList]);
+
+  const handleItemClick = (link: string) => {
+    navigateToUrl(link);
```

Comment:
does the link contains workspace information?  if not, we probably should append workspace id first. I remember @SuZhou-Joe have a PR to make recentlyAccess item as global

Reply:
Yeah, @raintygao could you please use the exported `createRecentNavLink` method from `src/core/public/chrome/ui/header/nav_link.tsx` to generate the final url?

Reply:
Sure, I will try it.